src/molecule/__init__.py

```diff
@@ -25,8 +25,8 @@
 __metaclass__ = type
 
 try:
-    import pkg_resources
+    import importlib_metadata
 
-    __version__ = pkg_resources.get_distribution("molecule").version
+    __version__ = importlib_metadata.version("molecule")  # type: ignore
 except Exception:
```

Comment:
I don't think that with importlib-metadata we should be having a catch-all handler. Let's do
```suggestion
except importlib_metadata.PackageNotFoundError:
```